components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionConsumer.java

```diff
@@ -0,0 +1,368 @@
+                            channel.txCommit();
+                        } catch (IOException e) {
+                            log.error("Error while committing transaction", e);
+                        }
+                    } else {
+                        try {
+                            channel.txRollback();
+                        } catch (IOException e) {
+                            log.error("Error while trying to roll back transaction", e);
+                        }
+                    }
+                }
+            } else {
+                idle = true;
+            }
+        }
+
+    }
+
+    /**
+     * Create a queue consumer using the properties from inbound listener configuration
+     *
+     * @throws IOException on error
+     */
+    private void initConsumer() throws IOException {
+        factoryName = rabbitMQProperties.getProperty(RabbitMQConstants.RABBITMQ_CON_FAC);
+        log.debug("Initializing consumer for inbound " + factoryName);
+        connection = getConnection();
+        channel = connection.createChannel();
+        queueName = rabbitMQProperties.getProperty(RabbitMQConstants.QUEUE_NAME);
+        routeKey = rabbitMQProperties.getProperty(RabbitMQConstants.QUEUE_ROUTING_KEY);
+        exchangeName = rabbitMQProperties.getProperty(RabbitMQConstants.EXCHANGE_NAME);
+
+        String autoAckStringValue = rabbitMQProperties.getProperty(RabbitMQConstants.QUEUE_AUTO_ACK);
+        if (autoAckStringValue != null) {
+            try {
+                autoAck = Boolean.parseBoolean(autoAckStringValue);
+            } catch (Exception e) {
+                log.debug("Format error in rabbitmq.queue.auto.ack parameter");
+            }
+        }
+        //If no queue name is specified then inbound factory name will be used as queue name
+        if (StringUtils.isEmpty(queueName)) {
+            queueName = factoryName;
+            log.info("No queue name is specified for " + factoryName + ". " +
+                    "inbound factory name will be used as queue name");
+        }
+
+        if (routeKey == null) {
+            log.info("No routing key specified. Using queue name as the " + "routing key.");
+            routeKey = queueName;
+        }
+
+        if (!StringUtils.isEmpty(queueName)) {
+            //declaring queue
+            RabbitMQUtils.declareQueue(connection, queueName, rabbitMQProps);
+        }
+
+        if (!StringUtils.isEmpty(exchangeName)) {
+            //declaring exchange
+            RabbitMQUtils.declareExchange(connection, exchangeName, rabbitMQProps);
+
+            if (!channel.isOpen()) {
+                channel = connection.createChannel();
+                log.debug("Channel is not open. Creating a new channel for inbound " + factoryName);
+            }
+            channel.queueBind(queueName, exchangeName, routeKey);
+            log.debug("Bind queue '" + queueName + "' to exchange '" + exchangeName + "' with route key '" + routeKey + "'");
```

Comment:
Use log.isDebug enabled across all components when logging debug messages. that needs to be fixed across the entire PR.